crates/evm/src/lib.rs

```diff
@@ -262,17 +279,65 @@ pub struct NextBlockEnvAttributes {
     pub gas_limit: u64,
 }
 
-/// Function hook that allows to modify a transaction environment.
-pub trait TxEnvOverrides {
-    /// Apply the overrides by modifying the given `TxEnv`.
-    fn apply(&mut self, env: &mut TxEnv);
+/// Abstraction over transaction environment.
+pub trait TransactionEnv:
+    Into<revm_primitives::TxEnv> + Debug + Default + Clone + Send + Sync + 'static
+{
```

Comment:
this makes sense, this is basically our workaround revm's `Transaction` trait, because this only exposes things we care about so this should be something separate imo.